MIGRATION_GUIDE.md

```diff
@@ -1,4 +1,325 @@
+- [4.x to 5.x](#4x-to-5x)
+
+---
+
+# 5.x to 6.x
+
+Popper.js was updated to v2. The `instance.popperInstance` and `popperOptions`
+APIs have changed. You can
+[view its documentation here](https://popper.js.org/docs/v2/).
+
+## Imports
+
+`iife` and `cjs` were replaced with `umd` (Rollup chunking has been removed).
+
+## HTML Content
+
+To protect against XSS by default, `allowHTML` is now `false` by default. If
+you're passing strings of HTML to `content`, you must set `allowHTML: true`.
+
+## Themes
+
+### If you were creating custom themes
+
+<details>
+<summary>View details</summary>
+
+`.tippy-tooltip` has become `.tippy-box`, and theming is done via an attribute
+now, to match the other props.
+
+The following:
+
+```css
+.tippy-tooltip.tomato-theme {
+  background-color: tomato;
+}
+```
+
+Has become:
+
+```css
+.tippy-box[data-theme~='tomato'] {
+  background-color: tomato;
+}
+```
+
+> The `~=` attribute operator allows you to specify mutliple theme names like
+> before with class names.
+
+For `.tippy-arrow`, you'll need to specify its color on the `::before`
+pseudo-element.
+
+The following:
+
+```css
+.tippy-tooltip.tomato-theme[data-placement^='top'] > .tippy-arrow {
+  border-top-color: tomato;
+}
+```
+
+Has become:
+
+```css
+.tippy-box[data-theme~='tomato'][data-placement^='top'] > .tippy-arrow::before {
+  border-top-color: tomato;
+}
+```
+
+In addition, if you were altering the pixel values, it may need to be adjusted.
+
+</details>
+
+### If you were targeting `.tippy-popper`
+
+<details>
+<summary>View details</summary>
+
+`.tippy-popper` is no longer a selector, and is considered an implementation
+detail for the most part now. `[data-tippy-root]` attribute selector replaces it
+if necessary.
+
+</details>
+
+## Instance
+
+### If you were using `instance.popperChildren`
+
+<details>
+<summary>View details</summary>
+
+This no longer exists due to the user's ability to specify any structured DOM
+with `render()` (Headless Tippy).
+
+To access the `.tippy-box` element with the default render function
+(`.tippy-tooltip` in v5):
+
+```js
+const box = instance.popper.firstElementChild;
+```
+
+</details>
+
+## Methods
+
+### If you were using `.show()` or `.hide()` with a duration argument
+
+<details>
+<summary>View details</summary>
+
+These no longer take a duration argument. Instead, use
+`.setProps({duration: ...})` before calling them if necessary.
+
+To replicate `.hide(0)`:
+
+```js
+instance.hide();
+instance.unmount();
+```
+
+</details>
+
+## Props
+
+### If you were using `boundary`
+
+<details>
+<summary>View details</summary>
+
+Often, this was to solve a problem in Popper 1, where you set
+`boundary: "window"`. This is no longer necessary. If you'd like to change it
+anyway, you can set it in `popperOptions`:
+
+```js
+tippy(targets, {
+  popperOptions: {
+    modifiers: [
+      {
+        name: 'preventOverflow',
+        options: {
+          // `boundary` and `rootBoundary` are accepted here
```

Comment:
maybe just write it like this:

```js
{
  name: 'preventOverflow',
  options: {
    boundary: 'window'
  }
}
```

`rootBoundary` is new in popper v2 so it's not needed in migration guide.

(I was confused by the comment, so I had to look up the documentation, hence this suggestion)